test/Sema/diag_unintended_optional_behavior.swift

```diff
@@ -7,76 +7,44 @@ func takeAny(_ left: Any, _ right: Any) -> Int? {
 func throwing() throws -> Int? {}
 
 func warnOptionalToAnyCoercion(value x: Int?) -> Any {
-  let a: Any = x // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}{{17-17= ?? <#default value#>}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}{{17-17=!}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{17-17= as Any}}
+  let a: Any = x
 
   let b: Any = x as Any
 
-  let c: Any = takeAny(a, b) // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}
+  let c: Any = takeAny(a, b)
 
   let _: Any = takeAny(c, c) as Any
 
-  let _: Any = (x)  // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}
+  let _: Any = (x)
 
   let f: Any = (x as Any)
   let g: Any = (x) as (Any)
 
-  _ = takeAny(f as? Int, g) // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}
+  _ = takeAny(f as? Int, g)
```

Comment:
Casts here and later have to be diagnosed as well because `as?` would result in an optional type.